grpclb/src/test/java/io/grpc/grpclb/GrpclbLoadBalancerProviderTest.java

```diff
@@ -0,0 +1,159 @@
+    assertThat(config.getMode()).isEqualTo(Mode.PICK_FIRST);
+    assertThat(config.getTarget()).isNull();
+  }
+
+  @Test
+  public void retrieveModeFromLbConfig_roundRobin() throws Exception {
+    String lbConfig = "{\"childPolicy\" : [{\"round_robin\" : {}}, {\"pick_first\" : {}}]}";
+
+    ConfigOrError configOrError =
+        provider.parseLoadBalancingPolicyConfig(parseJsonObject(lbConfig));
+
+    assertThat(configOrError.getConfig()).isNotNull();
+    GrpclbConfig config = (GrpclbConfig) configOrError.getConfig();
+    assertThat(config.getMode()).isEqualTo(Mode.ROUND_ROBIN);
+    assertThat(config.getTarget()).isNull();
+  }
+
+  @Test
+  public void retrieveModeFromLbConfig_nullConfigUseRoundRobin() throws Exception {
+    ConfigOrError configOrError =
+        provider.parseLoadBalancingPolicyConfig(null);
+
+    assertThat(configOrError.getConfig()).isNotNull();
+    GrpclbConfig config = (GrpclbConfig) configOrError.getConfig();
+    assertThat(config.getMode()).isEqualTo(Mode.ROUND_ROBIN);
+    assertThat(config.getTarget()).isNull();
+  }
+
+  @Test
+  public void retrieveModeFromLbConfig_emptyConfigUseRoundRobin() throws Exception {
+    String lbConfig = "{}";
+
+    ConfigOrError configOrError =
+        provider.parseLoadBalancingPolicyConfig(parseJsonObject(lbConfig));
+
+    assertThat(configOrError.getConfig()).isNotNull();
+    GrpclbConfig config = (GrpclbConfig) configOrError.getConfig();
+    assertThat(config.getMode()).isEqualTo(Mode.ROUND_ROBIN);
+    assertThat(config.getTarget()).isNull();
+  }
+
+  @Test
+  public void retrieveModeFromLbConfig_emptyChildPolicyUseRoundRobin() throws Exception {
+    String lbConfig = "{\"childPolicy\" : []}";
+
+    ConfigOrError configOrError =
+        provider.parseLoadBalancingPolicyConfig(parseJsonObject(lbConfig));
+
+    assertThat(configOrError.getError()).isNotNull();
+    assertThat(configOrError.getError().getDescription())
+        .contains("childPolicy must be provided");
+  }
+
+  @Test
+  public void retrieveModeFromLbConfig_unsupportedChildPolicyUseRoundRobin()
+      throws Exception {
+    String lbConfig = "{\"childPolicy\" : [ {\"nonono\" : {}} ]}";
+
+    ConfigOrError configOrError =
+        provider.parseLoadBalancingPolicyConfig(parseJsonObject(lbConfig));
+
+    assertThat(configOrError.getError()).isNotNull();
+    assertThat(configOrError.getError().getDescription())
+        .contains("Unknown grpclb childPolicy: nonono");
+  }
+
+  @Test
+  public void retrieveModeFromLbConfig_skipUnsupportedChildPolicy() throws Exception {
+    String lbConfig = "{\"childPolicy\" : [ {\"nono\" : {}}, {\"pick_first\" : {} } ]}";
+
+    ConfigOrError configOrError =
+        provider.parseLoadBalancingPolicyConfig(parseJsonObject(lbConfig));
+
+    assertThat(configOrError.getError()).isNotNull();
+    assertThat(configOrError.getError().getDescription())
+        .contains("Unknown grpclb childPolicy: nono");
```

Comment:
This should use pick_first. TODO: make sure there is cross-language agreement on this behavior.

Ditto below.

Reply:
talked to @markdroth, c core is already doing what we want to do (skip unknown policy).